src/dll/Systems/ScriptSystem.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Hook.hpp"
+#include "ISystem.hpp"
+#include "Paths.hpp"
+#include "PluginBase.hpp"
+
+class ScriptSystem : public ISystem
+{
+public:
+    ScriptSystem(const Paths& aPaths);
+
+    ESystemType GetType() final;
+
+    void Startup() final;
+    void Shutdown() final;
+
+    bool Add(std::shared_ptr<PluginBase> aPlugin, const char* path);
+    std::wstring GetRedModArgs();
+    void SetScriptsBlobPath(RED4ext::CString*);
+    RED4ext::CString* GetScriptsBlobPath();
+    void SetUsingRedmod(bool);
+    bool IsUsingRedmod();
+    std::wstring CreatePathsFile();
+
+private:
+    bool _Add(std::shared_ptr<PluginBase> aPlugin, std::filesystem::path* path);
+    using Map_t = std::unordered_multimap<std::shared_ptr<PluginBase>, std::filesystem::path>;
```

Comment:
Is it necessary to keep track which plugins add the paths? If not I would just make this a `std::vector<path>`.

Reply:
I thought it might be nice for diagnostics - it won't be known to the user which plugins add which files (in the event of a redscript compilation error). We should print the plugins along with the file names in the log.

Reply:
Ahhh, that's nice!

Reply:
Yeap! I added that since :)